packages/lab/src/__tests__/__e2e__/date-picker/DatePicker.cy.tsx

```diff
@@ -0,0 +1,205 @@
+
+const formatDay = (date: DateValue) => {
+  return formatDate(date, {
+    day: "2-digit",
+    month: "long",
+    year: "numeric",
+  });
+};
+
+describe("GIVEN a DatePicker", () => {
+  checkAccessibility(composedStories);
+
+  describe("WHEN single datepicker is mounted", () => {
+    it("THEN it should mount with the specified defaultStartDate", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("textbox").should("have.value", formatInput(testDate));
+    });
+    it("THEN should format a valid date with a different format on blur", () => {
+      const onChange = (event: ChangeEvent<HTMLInputElement>) => {
+        // React 16 backwards compatibility
+        event.persist();
+      };
+      cy.mount(<Default defaultStartDate={testDate} onChange={onChange} />);
+      cy.findByRole("textbox").click().clear().type(testInput);
+      cy.findByRole("textbox").blur();
+      cy.findByRole("textbox").should(
+        "have.value",
+        formatInput(testDate.add({ days: 1 }))
+      );
+    });
+    it("THEN should error and not format invalid dates on blur", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("textbox").click().clear().type("date");
+      cy.findByRole("textbox").blur();
+      cy.findByRole("textbox").should("have.value", "date");
+    });
+    it("THEN clicking the calendar button should open the panel", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("button", { name: "Open Calendar" }).realClick();
+      cy.findByRole("application").should("exist");
+    });
+    it("THEN should close the calendar panel once a date is selected", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("button", { name: "Open Calendar" }).realClick();
+      cy.findByRole("button", {
+        name: formatDay(testDate.add({ days: 11 })),
+      }).should("exist");
+      cy.findByRole("button", {
+        name: formatDay(testDate.add({ days: 11 })),
+      }).realClick();
+      cy.findByRole("application").should("not.exist");
+    });
+    it("THEN should not open the calendar when disabled", () => {
+      cy.mount(<Default defaultStartDate={testDate} disabled />);
+      cy.findByRole("button").should("be.disabled");
+    });
+    it("THEN it should update the selected month when changing selected date", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("textbox").click().clear().type(testInput);
+      cy.findByRole("textbox").blur();
+      cy.findByRole("button", { name: "Open Calendar" }).realClick();
+      cy.findByRole("button", {
+        name: "Next Month",
+      }).focus();
+    });
+    it("THEN it should clear the date if an empty input is submitted", () => {
+      cy.mount(
+        <Default
+          defaultStartDate={testDate}
+          // @ts-ignore
+          CalendarProps={{ visibleMonth: testDate }}
```

Comment:
Is this a valid use case for our user? if so, `ts-ignore` may need to be fixed?

Reply:
📝